rustls/src/client/mod.rs

```diff
@@ -161,8 +161,16 @@ impl ClientConfig {
         // is PACKET_OVERHEAD.
         if let Some(x) = *mtu {
             use crate::msgs::fragmenter;
-            debug_assert!(x > fragmenter::PACKET_OVERHEAD);
-            self.mtu = Some(x - fragmenter::PACKET_OVERHEAD);
+            self.mtu = x.checked_sub(fragmenter::PACKET_OVERHEAD);
+            if self.mtu.is_none() {
+                if !cfg!(test) {
```

Comment:
First, the doc comment needs to be updated to describe the new behavior.

FWIW, I don't think we need to be this fancy. I suggest we stick to normal Rust patterns and just return a `Result<(), MtuTooSmallError>` where the caller can either choose to `unwrap()` it or deal with the error. Only advanced users of Rustls will be calling `set_mtu()` anyway.

Reply:
I did think about that, but the likelihood of doing anything productive with `MtuTooSmallError` seemed low. That seems fine by me, if others want to go that route. RFC 791 requires 68 byte MTUs not to be fragmented.

Reply:
> I did think about that, but the likelihood of doing anything productive with MtuTooSmallError seemed low.

Most likely they're `unwrap()` it or wrap it in their own error type and forward it back to their caller, and maybe log it if they feel like it. The callers already have to deal with a lot of "I can't do anything about this" errors during configuration in my experience, so they'd probably like to handle them all consistently. At least, as a user (but not of this function) that's what I prefer.